bookkeeper-server/src/main/java/org/apache/bookkeeper/server/http/service/AutoRecoveryStatusService.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.bookkeeper.meta.LedgerUnderreplicationManager;
+import org.apache.bookkeeper.meta.MetadataDrivers;
+import org.apache.commons.lang3.ObjectUtils;
+
+/**
+ * HttpEndpointService that handles Autorecovery status related http requests.
+ *
+ * <p></p>The GET method returns the current status of Autorecovery. The output would be like {"enabled" : true}.
+ *
+ * <p>The PUT method requires a parameter 'enabled', and enables Autorecovery if its value is 'true',
+ * and disables Autorecovery otherwise. The behaviour is idempotent if Autorecovery status is already
+ * the same as desired. The output would be the current status after the action.
+ *
+ */
+public class AutoRecoveryStatusService implements HttpEndpointService {
+    protected final ServerConfiguration conf;
+
+    public AutoRecoveryStatusService(ServerConfiguration conf) {
+        this.conf = conf;
+    }
+
+    @Override
+    public HttpServiceResponse handle(HttpServiceRequest request) throws Exception {
+        return MetadataDrivers.runFunctionWithLedgerManagerFactory(conf,
+                ledgerManagerFactory -> {
+                    try (LedgerUnderreplicationManager ledgerUnderreplicationManager = ledgerManagerFactory
+                            .newLedgerUnderreplicationManager()) {
+                        switch (request.getMethod()) {
+                            case GET:
+                                return handleGetStatus(ledgerUnderreplicationManager);
+                            case PUT:
+                                return handlePutStatus(request, ledgerUnderreplicationManager);
+                            default:
+                                return new HttpServiceResponse("Not found method. Should be GET or PUT method",
+                                        HttpServer.StatusCode.NOT_FOUND);
+                        }
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new UncheckedExecutionException(e);
+                    } catch (Exception e) {
+                        throw new UncheckedExecutionException(e);
+                    }
+                });
+    }
+
+    private HttpServiceResponse handleGetStatus(LedgerUnderreplicationManager ledgerUnderreplicationManager)
+            throws Exception {
+        String body = JsonUtil.toJson(ImmutableMap.of("enabled",
+                ledgerUnderreplicationManager.isLedgerReplicationEnabled()));
+        return new HttpServiceResponse(body, HttpServer.StatusCode.OK);
+    }
+
+    private HttpServiceResponse handlePutStatus(HttpServiceRequest request,
+                                                LedgerUnderreplicationManager ledgerUnderreplicationManager)
+            throws Exception {
+        Map<String, String> params = ObjectUtils.defaultIfNull(request.getParams(), Collections.emptyMap());
+        String enabled = params.get("enabled");
+        if (enabled == null) {
```

Comment:
Extra check if equals "true" or "false"?

Reply:
@mino181295 we are evaluating the value at line 92